providers/logingov/logingov.go

```diff
@@ -0,0 +1,434 @@
+// Package login.gov implements the OAuth2 protocol for authenticating users through login.gov.
+// This package should NOT be used as a reference implementation of an OAuth2 provider for Goth,
+// as login.gov uses a different flow than typical providers.
+package logingov
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"fmt"
+
+	"github.com/markbates/goth"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// Standard Claims http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
+	// fixed, cannot be changed
+	subjectClaim  = "sub"
+	expiryClaim   = "exp"
+	audienceClaim = "aud"
+	issuerClaim   = "iss"
+
+	UuidClaim          = "uuid"
+	EmailClaim         = "email"
+	EmailVerifiedClaim = "email_verified"
+	GivenNameClaim     = "given_name"
+	FamilyNameClaim    = "family_name"
+	PhoneClaim         = "phone"
+	BirthdateClaim     = "birthdate"
+	SsnClaim           = "social_security_number"
+	AddressClaim       = "address"
+
+	clockSkew = 10 * time.Second
+)
+
+// These vars are necessary for login.gov auth to work.
+// These default to the login.gov sandbox values
+//   * See the getEnv fn below.
+// AcrValues: Only accepts LOA1 or LAO3 urls.
+//   * See https://developers.login.gov/oidc/#authorization
+//   * Currently only /loa/1 and /ial/1 work, as per the gsa-tts slack.
+var (
+	acrValues    = getEnv("LOGINGOV_ACR_VALUES", "http://idmanagement.gov/ns/assurance/loa/1")
```

Comment:
I recommend avoiding getEnv in the library in case the application using this library needs to have multiple instances of this login provider (possibly an app has an area where they only need LOA1 capabilities, but then a section of it needs upgraded LOA3 capabilities). What you could instead do is hardcode your default of LOA1 here, and then have a setup function (see: https://github.com/markbates/goth/pull/276/files#diff-a86775ae2ccd437b48a63db0a7239c49R52 for an example where Gitea login provider needs different setup than provided in default interface) where acr and approved emails are passed in via argument, and save into the provider struct.

Reply:
Duly noted. I updated it to have a NewCustomisedURL() fn, and now that fn sets AcrValues / ApprovedEmails.